habitat-baselines/habitat_baselines/rl/hrl/hl/fixed_policy.py

```diff
@@ -23,39 +21,25 @@ class FixedHighLevelPolicy(HighLevelPolicy):
 
     _solution_actions: List[Tuple[str, List[str]]]
 
-    def __init__(self, config, task_spec_file, num_envs, skill_name_to_idx):
-        """
-        Initialize the `FixedHighLevelPolicy` object.
-
-        Args:
-            config: Config object containing the configurations for the agent.
-            task_spec_file: Path to the task specification file.
-            num_envs: Number of parallel environments.
-            skill_name_to_idx: Dictionary mapping skill names to skill indices.
-        """
-        with open(get_full_habitat_config_path(task_spec_file), "r") as f:
-            task_spec = yaml.safe_load(f)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        self._num_envs = num_envs
-        self._skill_name_to_idx = skill_name_to_idx
         self._solution_actions = self._parse_solution_actions(
-            config, task_spec, task_spec_file
+            self._pddl_prob.solution
         )
 
-        self._next_sol_idxs = torch.zeros(num_envs, dtype=torch.int32)
-
-    def _parse_solution_actions(self, config, task_spec, task_spec_file):
-        if "solution" not in task_spec:
-            raise ValueError(
-                f"The ground truth task planner only works when the task solution is hard-coded in the PDDL problem file at {task_spec_file}."
-            )
+        self._next_sol_idxs = torch.zeros(self._num_envs, dtype=torch.int32)
 
+    def _parse_solution_actions(self, solution):
         solution_actions = []
-        for i, sol_step in enumerate(task_spec["solution"]):
-            sol_action = parse_func(sol_step)
+        for i, sol_step in enumerate(solution):
+            sol_action = (
```

Comment:
let's call this variable 'hl_action' to distinguish from future 'll_action'

Reply:
Done.